src/BloomExe/web/FakeTempFile.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Bloom.web
+{
+	/// <summary>
+	/// A TempFile is designed to have a API somewhat like TempFile. However, instead
+	/// of an actual temp file, it wraps the lifespan during which a particular url can be used
+	/// to retrieve something from EnhancedImageServer. To improve compatibility, it has
+	/// a 'Path' property which is actualy the URL that should be given to a browser
+	/// (however, it is not a path to a real disk file).
+	/// </summary>
+	public class FakeTempFile : IDisposable
```

Comment:
I'm looking for a name that would be more descriptive. This doesn't subclass TempFile, and even if it did, "fake" just tells me what it *isn't*, not what it is. The key thing seems to be that it's a URL-holder that for some reason tells the server when it has been disposed. Maybe I'll come back here later after reading more.